src/Microsoft.Extensions.Diagnostics.HealthChecks.Abstractions/IHealthCheck.cs

```diff
@@ -0,0 +1,27 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Extensions.Diagnostics.HealthChecks
+{
+    /// <summary>
+    /// Represents a health check, which can be used to check the status of a component in the application, such as a backend service, database or some internal
+    /// state.
+    /// </summary>
+    public interface IHealthCheck
```

Comment:
`IHealthCheckSource` or something like that since it's the thing that does a check?

Reply:
If it's the thing that does the check, is it not a "Health Check" ? :)